### PR TITLE
First Level Carousel + Initial Testing

### PR DESCRIPTION
**Pull request by f3lixding**
Changes included in this PR:
- Changed the architecture for the main component (to allow for animation).
- Added testing for main component as well as component level for server. (the test probably isn't passing currently. I shall tweak it for my next PR).

### CODE REVIEW

File: client/components/App.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import App from './App';
+import sampleData from '../assets/sampleData';
+
+
+describe('Should mount with data ready to be displayed', () => {
+  const wrapper = shallow(<App sampleData={sampleData} />);
+  it('should have access to an array of data for homes', () => {
+    const span = wrapper.find('span');
+    expect(true).toBe(true);
```
Comment (MalanchukA) [inline client/components/App.test.js]: Is this line necessary?

File: client/components/App.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    super(props);
+    this.state = {
+      activeCardIndex: 1,
+      activeCard: this.props.sampleData[0],
+      cardArray: this.props.sampleData,
+    }
+  }
+
+  componentDidMount() {
+    // fetch request to populate data
```
Comment (MalanchukA) [inline client/components/App.js]: I don't quite understand what is happening inside componentDidMount

File: client/components/Card.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        price: this.props.listingDetail.price,
+        review: this.props.listingDetail.review,
+        title: this.props.listingDetail.title,
+        type: this.props.listingDetail.type
+      })
+    }
+  }
+
+  componentDidMount() {
+    // debugger;
```
Comment (MalanchukA) [inline client/components/Card.js]: Same here

File: client/components/App.test.js (hunk 1 above)
Comment (f3lixding) [reply client/components/App.test.js]: Yea that was a line left over from the example test. I shall delete it.

File: client/components/App.js (hunk 2 above)
Comment (f3lixding) [reply client/components/App.js]: I am going to complete the component with mock/sample data first. And then I will complete the lifecycle method with proper fetch methods in there.

File: client/components/Card.js (hunk 3 above)
Comment (f3lixding) [reply client/components/Card.js]: Same situation here.

**Pull request merged**